File: web/src/types/book.ts
export interface Book {
	id: string,
	title: string,
	author: string,
	genre: string,
	publishedAt: string,
	imageUrl: string,
	fav: boolean
}

export interface Pageable<T> {
	content: T[],
	totalElements: number,
	totalPages: number,
	page: number
}

File: service/src/main/java/com/example/service/entity/Book.java
package com.example.service.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@Entity(name = "book")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Book {

   @Id
   @GeneratedValue(strategy = GenerationType.IDENTITY)
   private Long id;

   @Column(nullable = false)
   private String title;
   @Column(nullable = false)
   private String author;
   @Enumerated(EnumType.STRING)
   @Column(nullable = false)
   private Genre genre;
   @Column(nullable = false)
   private LocalDate publishedAt;
   @Column(nullable = false)
   private String imageUrl;
   @Column(nullable = false)
   private boolean fav;
}
